### ISSUE TITLE
Precompile text flickers

### ISSUE THREAD

**Issue by alhirzel** [JuliaLang/Pkg.jl#2327]
Using Julia 1.6.0beta1, when installing packages that have a lot of dependencies (e.g. `]add Plots`), the parallel precompilation feature saves a lot of time! While it is running, however, the queue of packages being compiled "flickers". This occurs in `urxvt-unicode` on Arch Linux. I also tested using a virtual console (i.e. Ctrl+Alt+F2) and saw the text flicker as well. It seems that any time one of the status bullets changes, the entirety of the list of packages is redrawn.

It isn't a bug, more of a comment on possible last-mile polish for this wonderful feature. Thank you for your work!

**Comment by ianshmean** [comment]
Thanks for the report. We print the entire update as a single print command, including the ansi escape characters to overwrite the previous text block. That means that any blank screen/incomplete text time that would cause flickering will be caused by the inherent speed of the terminal, not julia speed.  So I don't believe we can do anything about it unfortunately.

**Comment by alhirzel** [comment]
https://github.com/JuliaLang/Pkg.jl/blob/aa1332c79545504694188f1eb529dd3650bebc68/src/API.jl#L1110

I think avoiding `ansi_cleartoend` here would help, but some bookkeeping of screen contents will be required. When the order/position of items being printed has not changed and only the `anim_char` has, perhaps overwriting the `anim_char` would be sufficient.

I usually don't dive this deep into the ANSI escape sequences; perhaps some nuance is lost on my naivety. :)  Thanks again for looking at this!

**Comment by ianshmean** [comment]
Just updating the characters that change is an interesting idea.. but I imagine it could explode the complexity of the print loop. 

As an aside, perhaps that'd make for an interesting julia package..

**Comment by alhirzel** [comment]
Agreed, perhaps it would be related to https://github.com/JuliaLang/julia/blob/master/stdlib/REPL/src/TerminalMenus/AbstractMenu.jl

**Comment by KristofferC** [comment]
Ah, this was the text, not the bar itself. We can keep it open then. But I'll move it to the Julia repo since that is where the precompile code now resides.
